### PR TITLE
feat(map): add satellite overlay toggle

### PR DESCRIPTION
**Pull request by hgrsd**
This adds the satellite overlay toggle to this repo, which is the upstream for bellingcat's ukraine timemap, as discussed [here](https://github.com/bellingcat/ukraine-timemap/pull/12#issuecomment-1077580653).

Kapture 2022-03-24 at 18 12 39

### CODE REVIEW

**Comment by hgrsd** [comment]
@breezykermo as discussed, opening a PR here for the sat toggle

File: src/components/space/carto/Map.js [hunk 1]
```diff
@@ -91,9 +103,45 @@ class Map extends React.Component {
     }
   }
 
+  getTileUrl(tiles) {
```
Comment (hgrsd) [inline src/components/space/carto/Map.js:106]: This logic for getting the URL is functionally identical to what used to happen in `initializeMap`. I refactored the conditions ever so slightly to make it (in my view) easier to read.

File: src/components/space/carto/atoms/SatelliteOverlayToggle.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import mapImg from "../../../../assets/satelliteoverlaytoggle/map.png";
+import satImg from "../../../../assets/satelliteoverlaytoggle/sat.png";
+
+const SatelliteOverlayToggle = ({
+  switchToSatellite,
+  reset,
+  isUsingSatellite,
+}) => {
+  return (
+    <div id="satellite-overlay-toggle" className="satellite-overlay-toggle">
```
Comment (ebefarooqui) [inline src/components/space/carto/atoms/SatelliteOverlayToggle.js:13]: Could you condense this into one `<div>...<button />...</div>` using ternaries to set the appropriate props? (ie. `props = {className: isUsingSatellite ? "....toggle-map" : "....toggle-sat"}`)

File: src/components/space/carto/atoms/SatelliteOverlayToggle.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import copy from "../../../../common/data/copy.json";
+import { language } from "../../../../common/utilities";
+import mapImg from "../../../../assets/satelliteoverlaytoggle/map.png";
+import satImg from "../../../../assets/satelliteoverlaytoggle/sat.png";
+
+const SatelliteOverlayToggle = ({
+  switchToSatellite,
```
Comment (ebefarooqui) [inline src/components/space/carto/atoms/SatelliteOverlayToggle.js:8]: If you were to make a `toggleSatelliteView`, you could use that here (ie. `onClick={() => toggleSatelliteView(true || false)}`)

File: src/reducers/ui.js [hunk 4]
```diff
@@ -1,9 +1,28 @@
 import initial from "../store/initial.js";
 
-import {} from "../actions";
+import { USE_SATELLITE_TILES_OVERLAY, RESET_TILES_OVERLAY } from "../actions";
 
 function ui(uiState = initial.ui, action) {
-  return uiState;
+  switch (action.type) {
+    case USE_SATELLITE_TILES_OVERLAY:
```
Comment (ebefarooqui) [inline src/reducers/ui.js:7]: Same comment about reducing the two actions to one: toggleSatelliteView...

**Review by ebefarooqui — COMMENTED** [review]
Some minor comments about structure but other than that, looks great! Thanks for making the feature and providing succinct tests.

**Comment by hgrsd** [comment]
Thanks for your review @ebefarooqui and for the suggestion, was a good simplification. 

I started out thinking we'd want to support more map layers, but when narrowed the feature down to just a satellite toggle. The code hadn't kept up!

**Comment by breezykermo** [comment]
Looks good, thanks @hgrsd!

**Pull request merged**